src/pal/src/loader/module.cpp

```diff
@@ -1339,29 +1339,33 @@ static HMODULE LOADLoadLibrary(LPCSTR shortAsciiName, BOOL fDynamic)
     // See if file can be dlopen()ed; this should work even if it's already loaded
     {
         // See GetProcAddress for an explanation why we leave the PAL.
-        PAL_LeaveHolder holder;
-        dl_handle = dlopen(shortAsciiName, RTLD_LAZY | RTLD_NOLOAD); 
-        if (!dl_handle)
+        PAL_LeaveHolder holder; 
+        
+        // P/Invokes are often declared with variations on the actual library name.
+        // For example, it's common to leave off the extension/suffix of the library
+        // even if it has one, or to leave off a prefix like "lib" even if it has one
+        // (both of these are done typically to smooth over cross-platform differences). 
+        // We try to dlopen with such variations on the original.
+        const char* const formatStrings[4] = // used with args: PAL_SHLIB_PREFIX, shortAsciiName, PAL_SHLIB_SUFFIX
         {
-            dl_handle = dlopen(shortAsciiName, RTLD_LAZY);
-        }
-
-        // P/Invoke calls are often defined without an extension in the name of the 
-        // target library. So if we failed to load the specified library, try adding
-        // a proper extension and load the library again.
-        if (!dl_handle)
+            "%s%s%s",     // prefix+name+suffix
+            "%.0s%s%.0s", // name
```

Comment:
My point for using the prefix / suffix first was to promote the pure name without any prefix / suffix as a recommended de-facto standard. The reason is that it would work well in cases the app is using a native lib that's available on multiple platforms, but obviously have different prefix and suffix on each.